src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -1163,6 +1186,26 @@ private NameValueCollection GetResolvedOptions(List<string> options)
             return dnsConnectionString._allOptions;
         }
 
+        private void MergeSslOptions()
```

Comment:
Should be called `ValidateSslAndTlsOptions`.

I don't think we should be "merging" them. We should simply remember what was set on the connection string and some other level will decide which ones are equivalent to each other.